rudder-core/src/main/scala/com/normation/rudder/reports/ComplianceMode.scala

```diff
@@ -44,23 +44,66 @@ import net.liftweb.json._
  * - error_only: only report for repaired and error reports.
  */
 
-sealed trait ComplianceMode {
-  def name: String
+sealed trait ComplianceModeName {
+  val name : String
 }
 
-final case object FullCompliance extends ComplianceMode {
+final case object FullCompliance extends ComplianceModeName {
   val name = "full-compliance"
 }
-final case class ChangesOnly (
-  heartbeatPeriod : Int
-)extends ComplianceMode {
-  val name = ChangesOnly.name
-}
 
-object ChangesOnly {
+final case object ChangesOnly extends ComplianceModeName {
   val name = "changes-only"
 }
 
+final case object ReportDisabled extends ComplianceModeName {
+  val name = "report-disabled"
+}
+
+object ComplianceModeName {
+  val allModes : List[ComplianceModeName] = FullCompliance :: ChangesOnly :: ReportDisabled :: Nil
+
+  def apply (value : String) : Box[ComplianceModeName] = {
```

Comment:
I'm not at ease with apply being able to fail - it looks to much like an object instanciation, and object instanciation failing are surprising (and a PITA to debug). 
All that to say: could you name you method "parse" ? (parse is totally expected to fail. In fact, most of the time, "parse" methods fail :)